log/log_root_signer.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
```

Comment:
Could you add the analogous methods for SignedMapRoot (in an appropriate package)? We already know we want those, so we might as well do it here.

Reply:
Can you suggest a top-level trillian directory name suitable to hold the analogous functions?
`map` would be lovely, but it's a golang reserved work :)

Maybe it would be better to have `SignLogRoot` and `SignMapRoot` as methods of trillian/crypto/signer.Signer - especially in the PR at present they'd both take `s *crypto.Signer` as the first argument.